code/bmpman/bmpman.cpp

```diff
@@ -2576,6 +2609,61 @@ void bm_page_in_start() {
 	gr_bm_page_in_start();
 }
 
+/**
+ * Stuffs Pathtypes with an additional path leading to a subdir of the same name as the given file (minus file extension).
+ * Used for allowing frames of .eff animations to be found in their own subdirectories.
+ *
+ * Example: if called with "debris.eff" and that file is in data/maps, then Pathtypes[CF_TYPE_TEMP_SUBDIR_LOOKUP].path
+ * will be stuffed with data/maps/debris, allowing data/maps/debris/debris_0000.dds (etc) to be found
+ */
+bool set_temp_subdir_pathtype(const char *filename) {
```

Comment:
Could this function be expanded so that it would allow using a subdirectory in an arbitrary path type? This is currently tied to a specific file which means that it's very limited in what it can do.